Add documentation on how to treat scalar & update openapi3 emitter (#1967)
microsoft/typespec: docs/extending-typespec/emitters-basics.md

In order to know where to emit files, the emitter context has a `emitterOutputDir` property that is automatically resolved using the `emitter-output-dir` built-in emitter options. This is set to `{cwd}/tsp-output/{emitter-name}` by default, but can be overridden by the user. Do not use the `compilerOptions.outputDir`

## Handling scalars

> Scalars are types in TypeSpec that most likely have a primitive or built-in datastructure represneting those in the target language.

Reviewer: ```suggestion
Scalars are types in TypeSpec that most likely have a primitive or built-in datastructure representing those in the target language.
```

Author: hhm weird, speelcheck should have caught that

Author: hhm I guess it is 2 valid words combined

Recommended logic for emitting scalar is to:

1. If scalar is a known scalar(e.g. `int32`), emit the known mapping.
2. Otherwise check scalar `baseScalar` and go back to `1.`
   2.1 After resolving which scalar apply any decorators

:::note

> If scalar is a more generic scalar that doesn't have a mapping(e.g. integer), recommendation is to subsitute with the next closest mapping(e.g. integer->int64) and emit a warning.

Reviewer: ```suggestion
If the scalar is generic and doesn't have a mapping (e.g. integer), we recommend substituting it with the next closest mapping (e.g. integer->int64) and emitting a warning.
```
